client/editor/editor.go

```diff
@@ -126,18 +126,142 @@ func (e *Editor) showPositions() {
 }
 
 // MoveCursor updates the Cursor position.
-func (e *Editor) MoveCursor(x, _ int) {
+func (e *Editor) MoveCursor(x, y int) {
+	// move cursor horizontally
 	newCursor := e.Cursor + x
 
-	if newCursor < 0 {
-		newCursor = 0
+	// move cursor vertically
+	if y > 0 {
+		newCursor = e.calcCursorDown()
+	}
+
+	if y < 0 {
+		newCursor = e.calcCursorUp()
 	}
+
 	if newCursor > len(e.Text) {
 		newCursor = len(e.Text)
 	}
+
+	if newCursor < 0 {
+		newCursor = 0
+	}
+
 	e.Cursor = newCursor
 }
 
+func (e *Editor) calcCursorUp() int {
+	pos := e.Cursor
+	// reset cursor if out of bounds
+	if pos > len(e.Text)-1 {
+		pos = len(e.Text) - 1
+	}
+
+	// if cursor is currently on newline, "move" it
+	if e.Text[pos] == '\n' {
+		pos--
+	}
+
+	cls := -1
```

Comment:
I'm sorry, but what does `cls` exactly do? Does it act like a placeholder for the current line length?

Reply:
Resolved in https://github.com/burntcarrot/rowix/pull/24#issuecomment-1367453114.